### PR TITLE
Fix a building for Qt Creator 4.6.0 (issue #84)

### PR DESCRIPTION
**Pull request by rdnvndr**

### CODE REVIEW

File: Parsers/CppParser/cppdocumentparser.cpp [hunk 1]
```diff
@@ -184,8 +184,12 @@ void CppDocumentParser::reparseProject()
...
     }
     CppTools::CppModelManager *modelManager = CppTools::CppModelManager::instance();
-    QStringList list = d->activeProject->files(ProjectExplorer::Project::SourceFiles);
-    d->filesInStartupProject = list.filter(d->cppRegExp);
+    QStringList list;
+    foreach (Utils::FileName fileName, d->activeProject->files(ProjectExplorer::Project::SourceFiles)) {
+        if (fileName.toString().contains(d->cppRegExp))
+            list << fileName.toString();
+    }
+    d->filesInStartupProject = list;
```
Comment (becrux) [inline Parsers/CppParser/cppdocumentparser.cpp]: Can you use d->filesInStartupProject directly? There's no use for the "list" variable here.

Comment (rdnvndr) [reply Parsers/CppParser/cppdocumentparser.cpp]: Impossible. It filters the list of filename.

Comment (becrux) [reply Parsers/CppParser/cppdocumentparser.cpp]: foreach (Utils::FileName fileName, d->activeProject->files(ProjectExplorer::Project::SourceFiles)) {
        if (fileName.toString().contains(d->cppRegExp))
            d->filesInStartupProject << fileName.toString();
}

Comment (rdnvndr) [reply Parsers/CppParser/cppdocumentparser.cpp]: It is better. Thanks @becrux.

**Comment by orgads** [comment]
Proposed another minimalistic approach: https://github.com/CJCombrink/SpellChecker-Plugin/pull/88

**Comment by CJCombrink** [comment]
@rdnvndr @becrux 
Thank you for the effort and the contribution, unfortunately I decided to use PR #88 contributed by @orgads 
Thus closing this issue.

**Pull request closed without merging**